src/cargo/core/resolver/context.rs

```diff
@@ -26,6 +26,12 @@ pub struct Context {
     pub activations: Activations,
     pub resolve_features: im_rc::HashMap<PackageId, Rc<HashSet<InternedString>>>,
     pub links: im_rc::HashMap<InternedString, PackageId>,
+    pub public_dependency:
+        im_rc::HashMap<PackageId, im_rc::HashMap<InternedString, (PackageId, bool)>>,
```

Comment:
We should probably already have this for all the other fields but could a comment be added here for this new field about what this is a map to/from?